python/ray/tune/tests/test_tuner.py

```diff
@@ -91,9 +91,9 @@ def test_tuner_with_xgboost_trainer(self):
         # prep_v1 = StandardScaler(["worst radius", "worst area"])
         # prep_v2 = StandardScaler(["worst concavity", "worst smoothness"])
         param_space = {
-            "scaling_config": {
-                "num_workers": tune.grid_search([1, 2]),
-            },
+            "scaling_config": ScalingConfig(
+                num_workers=tune.grid_search([1, 2])
+            ),
```

Comment:
This should fail without the support for dataclasses in https://github.com/ray-project/ray/pull/23607 right?